manticore/platforms/linux.py

```diff
@@ -1029,6 +1029,38 @@ def sys_chdir(self, path):
         except OSError as e:
             return e.errno
 
+    def sys_getcwd(self, buf, size):
+        '''
+        getcwd - Get the current working directory
+        :param int buf: Pointer to dest array
+        :param size: size in bytes of the array pointed to by the buf 
+        :return: buf (Success), or 0
+        '''
+        
+        try:
+            current_dir = os.getcwd()          
+            length = len(current_dir) + 1
+                    
+            if not os.path.isdir(current_dir):
+                logger.info("GETCWD: Working directory does not exist. Returning ENOENT")
+                return -errno.ENOENT
+                      
+            if size > 0 and size < length:
+                logger.info("GETCWD: size is greater than 0, but is smaller than the length"  
+                            "of the path + 1. Returning ERANGE")
+                return -errno.ERANGE
+        
+            if not (buf in self.current.memory and buf+length in self.current.memory):
```

Comment:
We actually provide an api for this check, you can rewrite the above check as:
`if not self.current.memory.access_ok(slice(buf, buf+length), 'w'):`